src/DotNetty.Buffers/ReadOnlyByteBufferStream.cs

```diff
@@ -23,24 +23,26 @@ public ReadOnlyByteBufferStream(IByteBuffer buffer, bool releaseReferenceOnClosu
 
         public override bool CanRead => true;
 
-        public override bool CanSeek => false;
+        public override bool CanSeek => true;
 
         public override bool CanWrite => false;
 
         public override long Length
         {
-            get { throw new NotSupportedException(); }
+            get => this.buffer.ReaderIndex + this.buffer.ReadableBytes;
```

Comment:
I remember you telling me this, but I didn't really see it in the code. Regardless, this is simpler/cleaner. I'll do it.